src/main/webapp/components/results/results.js

```diff
@@ -278,21 +326,24 @@ const Container = props => {
     )
   }
 
+  const onColumnUpdate = (columnOrder, columnHide) => {
```

Comment:
I think the updateUserPrefs function allows for passing in just the fields you want to update, which means you can call it with just {columnOrder, columnHide}. I noticed that the hook doesn't check to see if the userPrefs changed when the updateUserPrefs function is called. You could add that functionality into the hook if you wanted

Reply:
Changed to pass a map of user pref fields.

Not making changes for the hook to check if the userPrefs changed.
I don't think I have access to the currentPrefs outside of the update function which only deals with updating the cache,  not actually performing the mutation.

Reply:
I ended up making the changes for the hook to check for changes in userPrefs before writing.
Thanks for the suggested change, Sam.